app/src/main/kotlin/com/wire/android/ui/sharing/ImportMediaViewModel.kt

```diff
@@ -0,0 +1,382 @@
+                teamId = otherUser.teamId
+            )
+        }
+
+        else -> null // We don't care about connection requests
+    }
+
+    private fun searchShareableConversation(conversationDetails: List<ConversationItem>, searchQuery: String): List<ConversationItem> {
+        val matchingConversations =
+            conversationDetails.filter { details ->
+                when (details) {
+                    is ConversationItem.GroupConversation -> details.groupName.contains(searchQuery, true)
+                    is ConversationItem.PrivateConversation -> details.conversationInfo.name.contains(searchQuery, true)
+                    is ConversationItem.ConnectionConversation -> false
+                }
+            }
+        return matchingConversations
+    }
+
+    fun navigateBack() = viewModelScope.launch(dispatchers.main()) { navigationManager.navigateBack() }
+
+    suspend fun handleReceivedDataFromSharingIntent(activity: AppCompatActivity) {
+        val incomingIntent = ShareCompat.IntentReader(activity)
+        appLogger.e("Received data from sharing intent ${incomingIntent.streamCount}")
+        importMediaState = importMediaState.copy(isImporting = true)
+        when (incomingIntent.streamCount) {
+            0 -> {
+                // if stream count is 0 the type will be text, we check the type to double check if it is text
+                // todo : handle the text , we can get the text from incomingIntent.text
+            }
+
+            1 -> {
+                // ACTION_SEND
+                handleSingleIntent(incomingIntent, activity)
+            }
+
+            else -> {
+                // ACTION_SEND_MULTIPLE
+                handleMultipleActionIntent(activity)
+            }
+        }
+        importMediaState = importMediaState.copy(isImporting = false)
+    }
+
+    private suspend fun handleSingleIntent(incomingIntent: ShareCompat.IntentReader, activity: AppCompatActivity) {
+        incomingIntent.stream?.let { uri ->
+            incomingIntent.type?.let { mimeType ->
+                handleImportedAsset(activity, mimeType, uri)?.let { importedAsset ->
+                    importMediaState = importMediaState.copy(importedAssets = mutableListOf(importedAsset))
+                }
+            }
+        }
+    }
+
+    private suspend fun handleMultipleActionIntent(activity: AppCompatActivity) {
+        val importedMediaAssets = mutableListOf<ImportedMediaAsset>()
+        activity.intent.parcelableArrayList<Parcelable>(Intent.EXTRA_STREAM)?.forEach {
+            val fileUri = it.toString().toUri()
+            handleImportedAsset(activity, fileUri.getMimeType(activity).toString(), fileUri)?.let { importedAsset ->
+                importedMediaAssets.add(importedAsset)
+            }
+        }
+        importMediaState = importMediaState.copy(importedAssets = importedMediaAssets)
+    }
+
+    fun onImportedMediaSent() = viewModelScope.launch(dispatchers.default()) {
```

Comment:
I think this returns job? Do we want it ?

Reply:
No we don't. Should we?

Reply:
idk but I though that launch returns a job ? but maybe not if you use it with viewModelScope just asking to make sure if we are doing something intentional here :P

Reply:
It does return a job, we just don't really care about the variable containing it

Reply:
so does it make sense to return this job ?

Reply:
For now we don't use that returned job. It was just a way to save some lines of code, but I guess it doesn't harm either 🤷